### ISSUE TITLE
Why can't I reproduce the results?

### ISSUE THREAD

**Issue by tyler55427** [QianWangX/EditCLIP#3]
When I try to run the `src/sd_ip2p_train/eval_ip2p_transfer.py`, using the images in assets which are the default. I can not reproduce the result in the paper.

<img width="512" height="512" alt="Image" src="https://github.com/user-attachments/assets/aeb071e7-d951-49b3-8fb5-5ada329aa3bf" />

<img width="279" height="119" alt="Image" src="https://github.com/user-attachments/assets/d6be849d-0017-43db-b678-033aeba5c5ec" />

**Comment by tyler55427** [comment]
I use the pretrained weight to genearate the result above. 
Editclip from [https://huggingface.co/QWW/EditCLIP](https://huggingface.co/QWW/EditCLIP). 
Unet from [https://huggingface.co/QWW/EditCLIP-IP2P](https://huggingface.co/QWW/EditCLIP-IP2P). 
Do I need to retrain the model with the experimental setup described in the paper to obtain better results?

**Comment by QianWangX** [comment]
@tyler55427 Thank you. You don't need to retrain the model with the setup again. Could you please first try to set `resolution=512`? In the paper we train with resolution 256, but test with 512. However, I will also check again the pretrained weights and get back to you.

**Comment by tyler55427** [comment]
@QianWangX Thanks for your responding! I try to set `--resolution 512` and get a better result.
<img width="512" height="512" alt="Image" src="https://github.com/user-attachments/assets/a25f8429-d6db-4f2a-a2cc-0a6a33880806" />
This one forgot to resize the references.

<img width="512" height="512" alt="Image" src="https://github.com/user-attachments/assets/0c0f9119-39da-4bba-8220-d93d6558b521" />

After resizeing the references, I got this result.

But when I try to reproduce the metrics like CLIP-Diretion showed in the paper, I cannot obtain similar result. I use the evalution code in [https://github.com/timothybrooks/instruct-pix2pix/blob/main/metrics/clip_similarity.py](https://github.com/timothybrooks/instruct-pix2pix/blob/main/metrics/clip_similarity.py) and evaluate on the benchark [TOP-Bench](https://drive.google.com/file/d/1uhOA8r0yBwDXg0aUCdJsoY4smer0rOv_/view) mentioned in paper. 
I get LPIPS 0.294, but 0.233 in paper.
I get CLIP(CLIP-Score) 0.187, but 0.216 in paper.
I get CLIP Dir. 0.052, but 0.143 in paper.

Could you release the evalutaion code? I want to reproduce the results. Thank you!!!

**Comment by Gorluxor** [comment]
@tyler55427 The benchmark was created based on TOP-Bench, which you linked. But the actual benchmark is [Top-Bench-X hosted on HuggingFace](https://huggingface.co/datasets/Aleksandar/Top-Bench-X).

**Comment by tyler55427** [comment]
@Gorluxor Thank you so much for pointing that out! I actually didn't notice that and had difficulty in seeking for the descriptions. I replaced the descriptions with instructions to evaluate and got the wrong results. Now I use the descriptions provided in Top-Bench-X and get the similar results. 
Really appreciate your help!!!

**Comment by tyler55427** [comment]
LPIPS: 0.294, length: 1277
CLIP-Score: 0.220, length: 1277
CLIP-Direction: 0.170, length: 1277

The results listed above are what I got. It seems that it is still a little different from the results listed in paper.
Why?

**Comment by QianWangX** [comment]
@tyler55427  I updated the weight of `image_proj_model.pt` in EditCLIP-IP2P. Could you please try again? Also, we averaged the results for the images generated under seed 0 to seed 4. However, I think that overall the results you got are similar to our original results

**Comment by tyler55427** [comment]
@QianWangX I try again with the new weight of `image_proj_model.pt` and get a better results. 
<img width="512" height="512" alt="Image" src="https://github.com/user-attachments/assets/c596b6ad-a8c0-4ff2-8046-11dc2593a696" />
OK, I will use the results I got for future study. Thanks!
